src/headers/gpm.h

```diff
@@ -25,6 +25,18 @@
 
 #include <paths.h>              /* _PATH_VARRUN etc. */
 
+#ifdef HAVE_NCURSES_H
```

Comment:
The use of the curses header also raises the question of what to compile first in a bootstrap setting? Should ncurses be compiled before gpm or the other way round? At present, gpm is often compiled without curses being present during build, but with this change, we turn bootstrapping into a chicken&egg problem.

Reply:
Those are valid complains.
I haven't thought that gpm is an optional dependency of ncurses. So, it would be "Build ncurses without gpm support, build gpm, build ncurses with gpm support", but that's painful and very suboptimal.
My main problems is, of course `int Gpm_Wgetch();` resolves in C23 context to `int Gpm_Wgetch(void);` so any attempts to call it would error out due to too many arguments.
Not sure how to resolve it in this situation.

Reply:
I agree that this is non-trivial to resolve. Debian has [heavily patched](https://sources.debian.org/src/gpm/1.20.7-11/debian/patches/050_dont_link_libcurses/) gpm to avoid depending on curses and forward declares an incompatible `WINDOW` in `lib/libcurses.c` as a result (the hidden struct is different).

The most practical solution I encountered thus far is making `Gpm_Wgetch` consume a `void*`. Implicit casts will allow passing `WINDOW*`. You can no longer pass zero or more than one argument and you can no longer pass non-pointers. While it does not add the desired type precision, it is a slight improvement and makes stuff compile in C23.